File: scripts/analyze-logs.ts
/**
 * Standalone Claude log analysis script.
 *
 * Run this manually to trigger an immediate analysis:
 *   tsx scripts/analyze-logs.ts
 *
 * Or let the main process call it on an interval.
 * This script can also be invoked by Claude Code hooks for automated self-improvement.
 */

import { runAnalysis } from '../src/self-healing/log-analyzer.js';
import { defaultScannerConfig } from '../src/config.js';

const config = { ...defaultScannerConfig };

console.log('Running Claude log analysis...\n');

runAnalysis(config)
  .then(result => {
    if (!result) {
      console.log('Analysis skipped (see logs for reason)');
      process.exit(0);
    }

    console.log('\n═══ ANALYSIS COMPLETE ═══\n');
    console.log('Summary:', result.summary);
    console.log('\nTop Issues:');
    result.topIssues.forEach((issue, i) => console.log(`  ${i + 1}. ${issue}`));

    console.log('\nStrategy Insights:');
    result.strategyInsights.forEach(si => {
      console.log(`  [${si.strategy}] ${si.observation}`);
      console.log(`    → ${si.recommendation}`);
    });

    if (result.newStrategySuggestions.length > 0) {
      console.log('\nNew Strategy Suggestions from Claude:');
      result.newStrategySuggestions.forEach((s, i) => console.log(`  ${i + 1}. ${s}`));
    }

    console.log(`\nConfidence: ${result.confidenceLevel}`);
    console.log('\nParameter changes applied (see logs for details)');
  })
  .catch((err: unknown) => {
    console.error('Analysis failed:', err);
    process.exit(1);
  });

File: scripts/analyze_logs.py
"""Standalone Claude log analysis script."""

import sys
from dataclasses import asdict

sys.path.insert(0, ".")

from src.self_healing.log_analyzer import run_analysis
from src.config import default_scanner_config
from src.types import ScannerConfig

config = ScannerConfig(**asdict(default_scanner_config))

print("Running Claude log analysis...\n")

result = run_analysis(config)

if not result:
    print("Analysis skipped (see logs for reason)")
    sys.exit(0)

print("\n=== ANALYSIS COMPLETE ===\n")
print("Summary:", result.summary)
print("\nTop Issues:")
for i, issue in enumerate(result.topIssues, 1):
    print(f"  {i}. {issue}")

print("\nStrategy Insights:")
for si in result.strategyInsights:
    print(f"  [{si.strategy}] {si.observation}")
    print(f"    -> {si.recommendation}")

if result.newStrategySuggestions:
    print("\nNew Strategy Suggestions from Claude:")
    for i, s in enumerate(result.newStrategySuggestions, 1):
        print(f"  {i}. {s}")

print(f"\nHealth Score: {result.overallHealthScore}/100")
print("\nParameter changes applied (see logs for details)")
